common/app/navigation/DCRNavigation.scala

```diff
@@ -100,16 +100,40 @@ object Nav {
 
   implicit val writes = Json.writes[Nav]
 
+  private def recursiveEditionalise(links: Seq[NavLink], edition: Edition, depth: Int = 0): Seq[NavLink] = {
+    if (depth > 5) links
```

Comment:
Why 5? I was looking at this the other day, and couldn't see where we had more than 2... but I imagine I hadn't investigated deeply enough.

Reply:
Honestly I just picked this completely arbitrarily to avoid an infinite loop

Reply:
Ah,I misunderstood this. I'm assuming this will always run 5 times, just on an empty list?

If so would 
```suggestion
    if (links.children.isEmpty) links
```

Reply:
As a point - the deepest I can see the nav run is 3, which is `News > Education` and `Lifestyle > Money`.

Reply:
It shouldn't run more times than the deepest level (so 3), because after that the Seq's should be empty & the map is never called.

This limit of 5 was a safety measure in case we ever (accidentally) created a circular structure, it wouldn't completely crash the code (instead this would happen during the JSON parsing I imagine)

Reply:
Yea, that makes sense - apologies, the recursive method won't run if `links = Nil`. got it.

This guard does confuse things a little though, as it feels like it's guarding against a non-situation. If we want to keep it, it might be worth leaving a note as to why it's there, as it feels like business logic rather than error handling.